From 9d843ecc0198bf35645f77bfca07ed76f8c0a57a Mon Sep 17 00:00:00 2001
From: Xiaowen Liu <b37945@freescale.com>
Date: Tue, 6 Mar 2012 17:30:28 +0800
Subject: [PATCH 249/635] ENGR00176196 fix hdmi problem and implement overscan
 for second display.

fix hdmi hot plug problem and implement overscan for second display.

Signed-off-by: Xiaowen Liu <b37945@freescale.com>
---
 mx6/libgralloc_wrapper/framebuffer.cpp | 14 ++++++--------
 1 file changed, 6 insertions(+), 8 deletions(-)

diff --git a/mx6/libgralloc_wrapper/framebuffer.cpp b/mx6/libgralloc_wrapper/framebuffer.cpp
index 2598435..8e5c196 100755
--- a/mx6/libgralloc_wrapper/framebuffer.cpp
+++ b/mx6/libgralloc_wrapper/framebuffer.cpp
@@ -590,14 +590,12 @@ static int fb_perform(struct gralloc_module_t const* module,
 
         case OPERATE_CODE_CHANGE:
             switch(operation & 0x0fff) {
-                case OPERATE_CODE_CHANGE_RESOLUTION:
-                case OPERATE_CODE_CHANGE_COLORDEPTH:
-		    err = unMapFrameBuffer(ctx, pm);
-		    err = mapFrameBufferWithParam(pm, param);
-		    if(err >= 0) {
-			fb_device_init(pm, ctx);
-		    }
-                    break;
+                case OPERATE_CODE_CHANGE_ROTATION:
+                case OPERATE_CODE_CHANGE_OVERSCAN: {
+		    size_t fbSize = pm->framebuffer->size;
+		    void* addr = (void*)(pm->framebuffer->base);
+		    memset(addr, 0, fbSize);
+                    } break;
                 default:
                     LOGE("<%s, %d> invalide operate code %d!", __FUNCTION__, __LINE__, (int)operation);
                     err = -1;
-- 
1.8.0

